src/neptune/api/searching_entries.py

```diff
@@ -154,19 +154,19 @@ def iter_over_pages(
     searching_after = None
     last_page = None
 
-    total = 0
+    total = get_single_page(
+        limit=0,
+        offset=0,
+        **kwargs,
+    ).get("matchingItemCount", 0)
 
-    progress_bar = progress_bar if step_size >= total else None
+    progress_bar = False if total < step_size else progress_bar  # disable progress bar if only one page is fetched
```

Comment:
I'm not 100% sure but shouldn't it be `<=`?

Reply:
Done